src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    /// <summary>
+    /// CA1862: Prefer the StringComparison method overloads to perform case-insensitive string comparisons.
+    /// </summary>
+    public class RecommendCaseInsensitiveStringComparisonFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(RecommendCaseInsensitiveStringComparisonAnalyzer.RuleId);
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            CancellationToken ct = context.CancellationToken;
+            if (ct.IsCancellationRequested)
+            {
+                return;
+            }
```

Comment:
I have never seen this pattern in the start of RegisterCodeFixesAsync

Reply:
You're correct, me neither, but I was wondering about this yesterday. I recently worked on writing async code in dotnet/runtime and @stephentoub taught me some tricks.

@stephentoub what do you think about using the cancellation detection at the beginning of async methods to exit early? Do you know if there could be a reason not to use it in this repo?

Reply:
Does this even do the right thing?  The expectation of cancelable Task-returning methods is that, if they're canceled, the Task ends in a Canceled state... this is going to look to the caller like it completed successfully even if it completed due to cancellation without actually doing its job.

Let's delete this.  If it's really important to do an up-front check for cancellation, you can add `context.ThrowIfCancellationRequested();` here instead, but if we're not doing that elsewhere in the repo, we shouldn't do so here, either.

Reply:
One thing I'm remembering I was told is that the cancellation detection is usually only done inside public methods, which is the case here.

I also used it in line 109 (a private method), maybe I could remove that one.

Reply:
> One thing I'm remembering I was told is that the cancellation detection is usually only done inside public methods, which is the case here.

I'm not sure who told you what, but this isn't accurate.

Reply:
> I'm not sure who told you what, but this isn't accurate. It is not the case that public methods must check the supplied cancellationToken manually, nor is it the case that non-public methods can't.

You're correct, I apologize, I completely misremembered. The explanations you gave me were around exceptions, not cancellation tokens, and I was handling them incorrectly: https://github.com/dotnet/runtime/pull/70574

> Let's delete this. If it's really important to do an up-front check for cancellation, you can add context.ThrowIfCancellationRequested(); here instead, but if we're not doing that elsewhere in the repo, we shouldn't do so here, either.

Ok thanks. Will remove.